File: src/test/java/org/meetkt/display/domain/model/DisplayUnitTest.java
package org.meetkt.display.domain.model;

import org.junit.jupiter.api.Test;
import org.meetkt.basket.domain.model.Basket;
import org.meetkt.basket.domain.model.Item;
import org.meetkt.basket.domain.model.Items;
import org.meetkt.invoice.domain.model.Invoice;
import org.meetkt.catalogue.domain.model.Product;
import org.meetkt.catalogue.fixture.ProductFixture;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class DisplayUnitTest {

    @Test
    void shouldReturnItemsFromBasket() {
        Product product = ProductFixture.aProduct().withProductId("001").build();
        Items expectedItems = new Items(List.of(new Item(product)));
        Basket basket = Basket.empty();
        basket.add(product);

        Display display = new Display(basket);

        assertThat(display.allBasketItems()).isEqualTo(expectedItems);
    }

    @Test
    void shouldReturnTotalPriceOfAllItemsInBasket() {
        Basket basket = Basket.empty();
        basket.add(ProductFixture.aProduct().withProductId("001").withPriceInt(10).build());
        basket.add(ProductFixture.aProduct().withProductId("002").withPriceDouble(20.78).build());

        Display display = new Display(basket);

        assertThat(display.totalBasketPrice()).isEqualTo(new Price(30.78));
    }

    @Test
    void shouldDisplayInvoiceWithOneItemForABasket() {
        Basket basket = Basket.empty();
        basket.add(ProductFixture.aProduct().withProductId("001").withPriceInt(10).build());
        Display display = new Display(basket);

        Invoice invoice = display.invoice();

        assertThat(invoice.totalItems()).isEqualTo(1);
    }
}

File: src/test/kotlin/in/meetkt/display/domain/model/DisplayUnitTest.kt
package `in`.meetkt.display.domain.model

import `in`.meetkt.basket.domain.model.Basket
import `in`.meetkt.basket.domain.model.Item
import `in`.meetkt.basket.domain.model.Items
import `in`.meetkt.basket.domain.model.Price
import `in`.meetkt.catalogue.fixture.ProductFixture
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class DisplayUnitTest {

    @Test
    fun shouldReturnItemsFromBasket() {
        val product = ProductFixture.aProduct().withProductId("001").withPriceDouble(12.21).build()
        val expectedItems = Items(mutableListOf(Item(product)))
        val basket = Basket.empty()
        basket.add(product)

        val display = Display(basket)

        assertThat(display.allBasketItems()).isEqualTo(expectedItems)
    }

    @Test
    fun shouldReturnTotalPriceOfAllItemsInBasket() {
        val basket = Basket.empty()
        basket.add(ProductFixture.aProduct().withProductId("001").withPriceInt(10).build())
        basket.add(ProductFixture.aProduct().withProductId("002").withPriceDouble(20.78).build())

        val display = Display(basket)

        assertThat(display.totalBasketPrice()).isEqualTo(Price(30.78))
    }
}